analyzers/src/SonarAnalyzer.CSharp/Rules/LiteralsShouldNotBePassedAsLocalizedParameters.cs

```diff
@@ -93,12 +94,17 @@ private static void AnalyzeInvocations(SyntaxNodeAnalysisContext context)
         private static void AnalyzeAssignments(SyntaxNodeAnalysisContext context)
         {
             var assignmentSyntax = (AssignmentExpressionSyntax)context.Node;
-            if (context.SemanticModel.GetSymbolInfo(assignmentSyntax.Left).Symbol is IPropertySymbol propertySymbol
-                && IsLocalizable(propertySymbol)
-                && IsStringLiteral(assignmentSyntax.Right, context.SemanticModel)
-                && !CSharpDebugOnlyCodeHelper.IsCallerInConditionalDebug(assignmentSyntax, context.SemanticModel))
+            var assignmentMappings = assignmentSyntax.MapAssignmentArguments();
+            foreach (var assignmentMapping in assignmentMappings)
             {
-                context.ReportIssue(Diagnostic.Create(Rule, assignmentSyntax.GetLocation()));
+                if (context.SemanticModel.GetSymbolInfo(assignmentMapping.Left).Symbol is IPropertySymbol propertySymbol
+                    && IsLocalizable(propertySymbol)
+                    && IsStringLiteral(assignmentMapping.Right, context.SemanticModel)
+                    && !CSharpDebugOnlyCodeHelper.IsCallerInConditionalDebug(assignmentSyntax, context.SemanticModel))
+                {
+                    context.ReportIssue(Diagnostic.Create(Rule, assignmentMapping.Right.GetLocation()));
```

Comment:
This was `assignmentSyntax.GetLocation()` before and is `assignmentMapping.Right.GetLocation()` now. This moves the location for every existing reported diagnostic. I don't know if we treat such a change as a breaking change. If so, we may want to revert this. The reasoning behind this is the error message:
```cs
Property1 = "some text"; // Noncompliant {{Replace this string literal with a string retrieved through an instance of the 'ResourceManager' class.}}
//          ^^^^^^^^^^^
```
It calls out "this string literal", but before the whole assignment was squiggled.

Reply:
This is not a breaking change - You'll need to update any UT that checks the location and there should be also some changes in the ITs expected results (which is done with the run of a script). 

It makes sense to me as well as a change.

Reply:
I'm going to update the ITs soon.